toolkit/tools/pkg/imagecustomizerlib/customizeoverlayfs.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+)
+
+func enableOverlayFS(overlays *[]imagecustomizerapi.OverlayFS, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	if overlays == nil {
+		return nil
+	}
+
+	// Integrate overlayfs dracut module and overlay driver into initramfs img.
+	// Integrate systemd veritysetup dracut module into initramfs img.
+	overlayfsDracutModule := "overlayfs"
+	overlayDracutDriver := "overlay"
+	err = buildDracutModule(overlayfsDracutModule, overlayDracutDriver, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	err = updateGrubConfigForOverlayFS(imageChroot, overlays)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateGrubConfigForOverlayFS(imageChroot *safechroot.Chroot, overlays *[]imagecustomizerapi.OverlayFS) error {
+	var err error
+	var newArgsParts []string
+
+	// Dereference the pointer to get the slice
+	overlayFSConfigs := *overlays
```

Comment:
nit: Do the dereference in `enableOverlayFS` since that is where the value is being checked for null.

Reply:
Fixed.